erpnext/selling/doctype/sales_order/sales_order.py

```diff
@@ -300,10 +300,14 @@ def validate_delivery_date(self):
 		if self.order_type == "Sales" and not self.skip_delivery_note:
 			delivery_date_list = [d.delivery_date for d in self.get("items") if d.delivery_date]
 			max_delivery_date = max(delivery_date_list) if delivery_date_list else None
-			if (max_delivery_date and not self.delivery_date) or (
-				max_delivery_date and getdate(self.delivery_date) != getdate(max_delivery_date)
-			):
+			if max_delivery_date and not self.delivery_date:
 				self.delivery_date = max_delivery_date
+			elif max_delivery_date and getdate(self.delivery_date) != getdate(max_delivery_date):
+				frappe.throw(
+					_("Delivery Date should be {0}, which is the max of all items' delivery dates").format(
+						max_delivery_date
+					)
+				)
```

Comment:
This will throw error any time you change delivery note in child table? :eyes:

Throwing doesn't seem like a good solution. This field should just be read only if the user input is not considered. 

If user input IS considered then it should be applied on all child table rows. 

E.g:
- On client side code when this field changes update all item rows
- On server side if `has_value_changed` then apply it to all child table items.

Reply:
> This will throw error any time you change delivery note in child table?

Yep

> On client side code when this field changes update all item rows. On server side if has_value_changed then apply it to all child table items.

I doubt if that's the behaviour users would expect when updating the delivery date field.

So maybe read only is the way to go. @rohitwaghchaure @deepeshgarg007 any idea on why it's not read only? If no particular reason, I'll make it read only.